sdk/modelsrepository/Azure.Iot.ModelsRepository/src/Fetchers/IModelFetcher.cs

```diff
@@ -13,8 +13,8 @@ namespace Azure.Iot.ModelsRepository.Fetchers
     /// </summary>
     internal interface IModelFetcher
     {
-        Task<FetchResult> FetchAsync(string dtmi, Uri repositoryUri, CancellationToken cancellationToken = default);
+        Task<FetchResult> FetchAsync(string dtmi, Uri repositoryUri, DependencyResolutionOption? resolutionOption, CancellationToken cancellationToken = default);
```

Comment:
I don't think we should have these as optional, it might be easier to manage them if we make them required and pass in either the provided option or the default option from the caller.

Reply:
what do you think?

Reply:
Done.